content/reference/sql/server/replication.md

```diff
@@ -4,88 +4,253 @@ title: Replication
++----------------------------------+
+```
+
+And back to the replica.
+
+```
+$ dolt sql -q "select * from test"
++----+----+
+| pk | c1 |
++----+----+
+| 0  | 0  |
+| 1  | 1  |
+| 2  | 2  |
++----+----+
+$ dolt log -n 1
+commit i97i9f1a3vrvd09pphiq0bbdeuf8riid (HEAD -> main, origin/main) 
+Author: Tim Sehn <tim@dolthub.com>
+Date:  Mon Jul 11 16:48:37 -0700 2022
+
+        Inserted (2,2)
 
-```SQL
-SELECT DOLT_COMMIT('-am', 'message')
 ```
 
-### Pull (on read) to replica
+### Replicate all branches
 
-Read replicas are instantiated with a remote:
+Only one of  [`@@dolt_replicate_heads`](../version-control/dolt-sysvars.md#doltreplicateheads)  or [`@@dolt_replicate_all_heads`](../version-control/dolt-sysvars.md#doltreplicateallheads) can be set at a time. So I unset `@@dolt_replicate_heads` and set `@@dolt_replicate_all_heads`.
 
-```bash
-dolt sql -q "SET PERSIST @@GLOBAL.dolt_read_replica_remote = 'origin'"
+```
+read_replica $ dolt config --unset --local sqlserver.global.dolt_replicate_heads
+Config successfully updated.
+read_replica $ dolt config --add --local sqlserver.global.dolt_replicate_all_heads 1
+Config successfully updated.
 ```
 
-A complete replication setup requires a pull spec with either 1) a set
-of heads, or 2) all heads (but not both):
+Now I'm going to make a new branch on the master and insert a new value on it.
 
-```bash
-dolt sql -q "SET PERSIST @@GLOBAL.dolt_replicate_heads = 'main,feature1'"
-dolt sql -q "SET PERSIST @@GLOBAL.dolt_replicate_all_heads = 1'"
 ```
+$ dolt sql -q "call dolt_checkout('-b', 'branch1'); insert into test values (3,3); call dolt_commit('-am', 'Inserted (3,3)');"
++--------+
+| status |
++--------+
+| 0      |
++--------+
+Query OK, 1 row affected
++----------------------------------+
+| hash                             |
++----------------------------------+
+| 4nrcqks869nrg4jofvuk72iui6hbk1bs |
++----------------------------------+
+```
+
+The read replica now has the change when I try and read it.
 
-On the replica end, pulling is triggered by an SQL `START TRANSACTION`.
-The first query in a session automatically starts a transaction. Setting
-`autocommit = 1`, which begins every query with a transaction, is
-encouraged on read replicas for convenience.
+(Found buggy behavior with dolt_checkout on replica)
+
+```
+read_replica $ dolt sql -q "call dolt_checkout('branch1'); select * from test;"
++--------+
+| status |
++--------+
+| 0      |
++--------+
+error on line 1 for query  select * from test: replication failed: working set not found
+replication failed: working set not found
+read_replica $ dolt branch
+  branch1                                       	
+* main                                          	
+read_replica $ dolt checkout branch1
+Switched to branch 'branch1'
+read_replica $ dolt sql -q "select * from test;"
++----+----+
+| pk | c1 |
++----+----+
+| 0  | 0  |
+| 1  | 1  |
+| 2  | 2  |
+| 3  | 3  |
++----+----+
+```
 
 ### Auto-fetching
 
+(Need to test this)
```

Comment:
the above example depends on auto-fetching also

Reply:
OK.